clustering/ejb/WFLY-7628_Distributed_EJB_Timers.adoc

```diff
@@ -0,0 +1,168 @@
+=== QE Contacts
+
+* mailto:msvehla@redhat.com[Martin Svehla]
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+Clustering, EJB
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* Create a `TimerService` implementation that delegates to separate persistent vs transient `TimerService` implementations.
+* Implement Infinispan-based `TimerService` for persistent timers that manages timer execution based on the consistent hash of the timer identifier.
+** Make existing ``default-data-store`` attribute of the timer service resource optional.  Use distributable implementation when this attribute is undefined.
+* Add default and named persistent timer management capabilities to distributable-ejb subsystem.
+** Add named Infinispan timer management resource that provides the above capability.
+* Add an attribute to the timer service resource within the ejb3 subsystem to determine the server-wide default provider for transient timer management.
+* Add ability to enable distributed timer management (in lieu of a timer store) within ejb3 subsystem for management of persistent timers.
+* Existing timer service resource operations must continue to work when executed against the distributable `TimerService`.
+
+=== Nice-to-Have Requirements
+
+* Add default and named transient timer management capabilities to distributable-ejb subsystem.
+* Add an attribute to the timer service resource within the ejb3 subsystem to determine the server-wide default provider for transient timer management.
+* Add ability to switch between distributed timer management or existing in-memory implementation for management of transient timers
+** This has the advantage of supporting passivation of timer instances to support applications with a sufficiently large number of transient timers in excess of some configured maximum.
+* Add support to timer service deployment descriptor namespace for overriding persistent/transient timer management per EJB.
+
+=== Non-Requirements
+
+* This feature will not change the default timer service behavior or deprecate any existing timer service mechanisms.
+
+== Backwards Compatibility
+
+* Servers started using the default ejb3 subsystem configuration will continue to behave as normal, using the existing in-memory `TimerService` implementation for both transient and persistent timers.
+
+=== Default Configuration
+
+The following caches will be added to the default `ejb` cache-container configuration with the Infinispan subsystem:
+
+For the HA profiles:
+
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent:add()
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent/component=locking:add(isolation=PESSIMISTIC)
```

Comment:
I see that the isolation level is set to `REPEATABLE_READ` in the docs and in the remote Infinispan integration test. What is the correct value for this configuration setting?